Physical_Engine/Query/Height.cs

```diff
@@ -0,0 +1,75 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2021, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System;
+using System.Linq;
+using BH.oM.Reflection.Attributes;
+using System.ComponentModel;
+using BH.oM.Physical.Elements;
+using BH.Engine.Geometry;
+
+namespace BH.Engine.Physical
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /****               Public Methods              ****/
+        /***************************************************/
+
+        [Description("Returns the vertical orthogonal height of a generic opening based on global coordinates of its BoundingBox.")]
+        [Input("opening", "A generic Opening object to query its height.")]
+        [Output("height", "The total height of the generic opening.")]
+        public static double IHeight(this IOpening opening)
+        {
+            return Height(opening as dynamic);
+        }
+
+        /***************************************************/
+
+        [Description("Returns the vertical orthogonal height of a door object based on global coordinates of its BoundingBox.")]
+        [Input("door", "A door object to query its height.")]
+        [Output("height", "The total height of the door object.")]
+        public static double Height(this Door door)
+        {
+            if(door == null)
+            {
+                BH.Engine.Reflection.Compute.RecordError("Cannot query the height of a null door.");
+                return 0;
+            }
+
+            return BH.Engine.Geometry.Query.Height(door.Location.IBounds());
```

Comment:
By adding the using of Spatial_Engine, you could query the bounds straight from the door. That is just details though, and I actually think what you have here might be quicker anyway.